std/experimental/allocator/building_blocks/aligned_block_list.d

```diff
@@ -0,0 +1,692 @@
+// Written in the D programming language.
+/**
+Source: $(PHOBOSSRC std/experimental/allocator/building_blocks/_aligned_block_list.d)
+*/
+module std.experimental.allocator.building_blocks.aligned_block_list;
+
+import std.experimental.allocator.common;
+import std.experimental.allocator.building_blocks.null_allocator;
+
+// Common function implementation for thread local and shared AlignedBlockList
+private mixin template AlignedBlockListImpl(bool isShared)
+{
+    import std.traits : hasMember;
+    import std.typecons : Ternary;
+
+    static if (isShared)
+    import core.internal.spinlock : SpinLock;
+
+private:
+    // Doubly linked list of 'AlignedBlockNode'
+    // Each node contains an `Allocator` followed by its payload
+    static struct AlignedBlockNode
+    {
+        AlignedBlockNode* next, prev;
+        Allocator bAlloc;
+
+        static if (isShared)
+        {
+            shared(size_t) bytesUsed;
+            // Since the lock is not taken when allocating, this acts like a refcount
+            // keeping the node alive
+            uint keepAlive;
+        }
+        else
+        {
+            size_t bytesUsed;
+        }
+    }
+
+    // Root of the internal doubly linked list
+    AlignedBlockNode* root;
+
+    // Number of active nodes
+    int numNodes;
```

Comment:
`uint`

Reply:
Wouldn't `size_t` be better?

Reply:
`size_t` is size of objects in memory. For counts we can use that of course, but just incidentally.